File: wk2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string para);
int count_words(string para);
int count_sentences(string para);

int main(void)
{
    string text = get_string("Text: \n");
    float nletters = count_letters(text);
    float nwords = count_words(text);
    float nsentences = count_sentences(text);

    float L = (100 / nwords) * nletters;
    float S = (100 /  nwords) * nsentences;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if(index > 16)
    {
        printf("Grade 16+\n");
    }
    else if(index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int)round(index));
    }

}

int count_letters(string para)
{
    int letters = 0;
    for( int i = 0, n = strlen(para); i < n; i++)
    {
        if(isalpha(para[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string para)
{
    int words = 0;
    for( int i = 0, n = strlen(para); i < n; i++)
    {
        if(isspace(para[i]))
        {
            words++;
        }
    }
    return words + 1;
}

int count_sentences(string para)
{
    int sentences = 0;
    for( int i = 0, n = strlen(para); i < n; i++)
    {
        if(para[i] == '.' || para[i] == '!' || para[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: wk6/sentimental-readability/readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")
    letters = 0
    sentences = 0
    words = 1
    for c in text:
        if c.isalpha():
            letters += 1
        elif c.isspace():
            words += 1
        elif c == "!" or c == "?" or c == ".":
            sentences += 1
    index(letters, words, sentences)

def index(letters, words, sentences):
    L = (100/words) * letters
    S = (100/words) * sentences
    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index < 1:
        print("Before Grade 1\n")
    elif index >= 16:
        print("Grade 16+\n")
    else:
        print(f"Grade {index}\n")



if __name__ == "__main__":
    main()